src/lib/services/terminate-service.ts

```diff
@@ -24,3 +25,42 @@ export async function terminateWorkflow({
     notifyOnError: false,
   });
 }
+
+type BulkTerminateWorkflowOptions = {
+  namespace: string;
+  workflowExecutions: WorkflowExecution[];
+  reason: string;
+};
+
+export async function batchTerminateWorkflows({
+  namespace,
+  workflowExecutions,
+  reason,
+}: BulkTerminateWorkflowOptions): Promise<null> {
+  const route = await routeForApi('workflows.batch.terminate', { namespace });
+
+  const runIds = workflowExecutions.map((wf) => wf.runId);
+  const query = runIds.reduce((queryString, id, index, arr) => {
+    queryString += `RunId="${id}"`;
+    if (index !== arr.length - 1) {
+      queryString += '+or+';
+    }
+
+    return queryString;
+  }, '');
+
+  return await requestFromAPI<null>(route, {
+    options: {
+      method: 'POST',
+      body: stringifyWithBigInt({
+        jobId: uuidv4(),
+        namespace,
+        visibilityQuery: query,
+        reason,
+        terminationOperation: {},
```

Comment:
Idk if we need to include `details` or `identity` here, but it seems to work fine without it.

Reply:
We don't need to, but down the road we might send 'Web UI' or something as identity for all APIs. But not for this PR